src/helpers/messaging.rs

```diff
@@ -160,19 +161,18 @@ pub struct GatewayConfig {
 }
 
 impl Gateway {
-    pub fn new<N: Network>(role: Role, network: &N, config: GatewayConfig) -> Self {
+    pub async fn new<T: Transport>(role: Role, network: Network<T>, config: GatewayConfig) -> Self {
         let (recv_tx, mut recv_rx) = mpsc::channel::<ReceiveRequest>(config.recv_outstanding);
         let (send_tx, mut send_rx) = mpsc::channel::<SendRequest>(config.send_outstanding);
-        let mut message_stream = network.recv_stream();
-        let mut network_sink = network.sink();
+        let mut message_stream = network.recv_stream().await;
 
         let control_handle = tokio::spawn(async move {
             const INTERVAL: Duration = Duration::from_secs(3);
 
             let mut receive_buf = ReceiveBuffer::default();
             let mut send_buf = SendBuffer::new(config.send_buffer_config);
-
-            let sleep = ::tokio::time::sleep(INTERVAL);
+            let mut pending_sends = FuturesUnordered::new();
```

Comment:
`send` may block if flow control is in place. this allows event loop to continue serving other traffic if send is blocked. There will be at most one in-flight send